query.go

```diff
@@ -940,6 +944,9 @@ func (qm *QueryMatches) Next() *QueryMatch {
 
 // Next will return the next match in the sequence of matches, as well as the index of the capture.
 //
+// Subsequent calls to [QueryCaptures.Next] will reuse the same memory,
+// this means that you need to copy/take out what you need from the [QueryMatch] before calling [QueryCaptures.Next] again.
+//
```

Comment:
```suggestion
// Subsequent calls to [QueryCaptures.Next] will overwrite the memory at the same location as prior matches, since the memory is reused.
// If you need to keep the data of a prior match without it being overwritten, you should copy what you need before calling [QueryCaptures.Next] again.
//
```